test/redis_test.rb

```diff
@@ -243,6 +245,49 @@ def test_query_instrumentation
     Semian.unsubscribe(subscriber)
   end
 
+  def test_changes_timeout_when_half_open_and_configured
```

Comment:
This test only tests the connect or read timeout, but most likely not both (which one requires some more intimacy with the gem, I'm not sure if it kills the connection after a timeout--some clients do). Most likely it maintains the connection and you 'just' need another test that'll disconnect before your second toxiproxy.

Reply:
Amazing suggestion.